mailpoet/lib/Statistics/StatisticsWooCommercePurchasesRepository.php

```diff
@@ -46,4 +46,25 @@ public function createOrUpdateByClickDataAndOrder(StatisticsClickEntity $click,
     $statistics->setSubscriber($click->getSubscriber());
     $this->flush();
   }
+
+  public function getRevenuesByCampaigns() {
+    $revenueStatsTable = $this->entityManager->getClassMetadata(StatisticsWooCommercePurchaseEntity::class)->getTableName();
+    $newsletterTable = $this->entityManager->getClassMetadata(NewsletterEntity::class)->getTableName();
+
+    return $this->entityManager->getConnection()->executeQuery('
+      SELECT
+        SUM(swp.order_price_total) AS revenue,
+        COALESCE(n.parent_id, n.id) AS campaign_id,
+        (CASE WHEN n.type = :notification_history_type THEN :notification_type ELSE n.type END) AS campaign_type,
+        COUNT(order_id) as orders_count
+      FROM ' . $revenueStatsTable . ' swp
+        JOIN ' . $newsletterTable . ' n ON
+          n.id = swp.newsletter_id
+          AND swp.click_id IN (SELECT MIN(click_id) FROM ' . $revenueStatsTable . ' ss GROUP BY order_id)
```

Comment:
Related to the comment [higher above](https://github.com/mailpoet/mailpoet/pull/4665/commits/2882171d8e606075ae7154ab7e0de5d951d53136#diff-6dca911f02a3f27ae96ba1f6af9bf490665746f2fca55456aca037726bbedd39R20), what you're doing here is to count each purchase ever only once, is that so?

I know that originally, the task was specified as tracking each purchase at most once per newsletter, which means it can be counted multiple times. Am I correct to assume the subquery wants to prevent that?

In any case, I think we should add a short comment explaining that. E.g.:

```
// The "SELECT MIN(click_id)..." subquery is used to count each purchase only once.
// In the UI, we count a purchase to multiple newsletters if clicks from multiple newsletters occured.
```

Reply:
> Am I correct to assume the subquery wants to prevent that?

Correct. I send totals per campaign and later want to aggregate using e.g. using campaign type. So I want to make sure an order is counted only once.